lockup/src/lib.rs

```diff
@@ -1725,6 +1725,163 @@ mod tests {
+        );
+        testing_env!(context.clone());
+
+        assert_eq!(
+            contract.get_vesting_information(),
+            VestingInformation::VestingHash(
+                VestingScheduleWithSalt {
+                    vesting_schedule: vesting_schedule.clone(),
+                    salt: SALT.to_vec().into(),
+                }
+                    .hash()
+                    .into()
+            )
+        );
+
+        // All the tokens are still locked
+        assert_eq!(contract.get_owners_balance().0, 0);
+        assert_eq!(contract.get_liquid_owners_balance().0, 0);
+        assert_eq!(contract.get_locked_amount().0, lockup_amount);
+
+        // Some tokens are vested, but they are still locked due to lockup
+        let locked_vested_amount_at_termination_day = contract
+            .get_locked_vested_amount(vesting_schedule.clone())
+            .0;
+        assert!(locked_vested_amount_at_termination_day > 0);
+
+        let unvested_amount_at_termination_day =
+            contract.get_unvested_amount(vesting_schedule.clone()).0;
+        assert!(unvested_amount_at_termination_day > 0);
+
+        // Terminate the vesting
+        context.predecessor_account_id = account_foundation();
+        context.signer_account_pk = public_key(3).into();
+        context.is_view = false;
+        testing_env!(context.clone());
+        contract.terminate_vesting(Some(VestingScheduleWithSalt {
+            vesting_schedule: vesting_schedule.clone(),
+            salt: SALT.to_vec().into(),
+        }));
+
+        context.is_view = true;
+        testing_env!(context.clone());
+        assert_eq!(
+            contract.get_vesting_information(),
+            VestingInformation::Terminating(TerminationInformation {
+                unvested_amount: unvested_amount_at_termination_day.into(),
+                status: TerminationStatus::ReadyToWithdraw,
+            })
+        );
+
+        // Moving further, to the moment when unlocking has already started
+        context = get_context(
+            system_account(),
+            to_yocto(LOCKUP_NEAR),
+            0,
+            to_ts(GENESIS_TIME_IN_DAYS + 400),
+            true,
+        );
+        testing_env!(context.clone());
+
+        let locked_today = contract.get_locked_amount().0;
+
+        // Moving next day
+        context = get_context(
+            system_account(),
+            to_yocto(LOCKUP_NEAR),
+            0,
+            to_ts(GENESIS_TIME_IN_DAYS + 401),
+            true,
+        );
+        testing_env!(context.clone());
+
+        let locked_tomorrow = contract.get_locked_amount().0;
+        assert!(
+            locked_tomorrow < locked_today,
+            "Locked amount should decrease"
+        );
+
+        // Moving further, to the moment when unlocking should stop
+        context = get_context(
+            system_account(),
+            to_yocto(LOCKUP_NEAR),
+            0,
+            to_ts(GENESIS_TIME_IN_DAYS + 720),
```

Comment:
Why 720? I feel it should stop unlocking the tokens vested over `YEAR - vesting_cliff_offset_in_days + termination_offset_in_days` (YEAR - 100 + 200 = 465 days of vested tokens out of 4 years vesting contract), and that should happen on `GENESIS_TIME_IN_DAYS + lockup_duration + 465`th day (given that lockup_duration is 365, I expect it to be +830) in lockup contract v3 implementation (without lockup cliff), and on `GENESIS_TIME_IN_DAYS + 465`th day in lockup contract v2 implementation (where lockup had cliff).

What do I miss?